Remove react proptypes references

rizoom opened the pull request:
Following on from #9757, the purpose of this PR is to remove the few remaining React.PropTypes references.
I have only found a few occurrences in the doc and the warnings messages in the tests so it is only of minor importance.

Commits related to the live editor and the "warning" page have been tested by running the documentation in local.

Commit related to the warning sentences in the tests is more problematic because it breaks the tests if the warning message in the [prop-types](https://github.com/facebook/prop-types) library is not updated as well.
If it is acceptable, another option would be to only test the beginning of the warning message.
I am unsure but I also guess that the _scripts/error-codes/codes.json_ file should be regenerated.
So maybe this last commit could be excluded from the PR, because given these constraints it may not be worth it.

gaearon commented:
>Commit related to the warning sentences in the tests is more problematic because it breaks the tests if the warning message in the prop-types library is not updated as well.

Can you split that into a separate PR, and then send a corresponding PR to `prop-types`? Thanks.

rizoom commented:
Sure, I will do that soon.

rizoom commented:
> Can you split that into a separate PR, and then send a corresponding PR to prop-types? Thanks.

I removed the commit that made the tests to fails, so this PR is now passing the tests.

Regarding the separate PR in this repository + prop-types. I updated prop-types and ran the tests in local but in fact the warning is being thrown by the create-react-class package (see `create-react-class#validateTypeDef`). Note that I am unable to find the create-react-class on github, so I am not sure that I can submit a PR against it.
**Summary**: The updated warning sentence in the React unit tests exists and can be thrown both by `prop-types` and `create-react-class`, but it seems that react unit tests are checking the warning thrown by `create-react-class` only.

rizoom commented:
@gaearon I just saw in another PR discussion that `create-react-class` is in the `15.6-dev`, so maybe I should submit the PR against that branch?
